### PR TITLE
feat(quantic): quantic smart snippet suggestions component created

### PR DESCRIPTION
**Pull request by mmitiche**
[SVCC-184](https://coveord.atlassian.net/browse/SVCC-184)

- Quantic Smart Snippet Suggestions created!

#### Demo:

https://user-images.githubusercontent.com/86681870/219024759-cfdac9e3-2192-4a8a-9938-f483947e9b27.mov

[SVCC-184]: https://coveord.atlassian.net/browse/SVCC-184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+function smartSnippetSuggestionsExpectations(
+  selector: SmartSnippetSuggestionsSelector
+) {
+  return {
+    displaySmartSnippetSuggestionsCard: (display: boolean) => {
+      selector
+        .smartSnippetSuggestionsCard()
+        .should(display ? 'exist' : 'not.exist')
+        .log(`${should(display)} display the smart snippet suggestions`);
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts:79]:
```suggestion
        .logDetail(`${should(display)} display the smart snippet suggestions`);
```

File: packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    },
+
+    displaySmartSnippetSuggestionsQuestion: (index: number, value: string) => {
+      selector
+        .smartSnippetSuggestionHeading(index)
+        .contains(value)
+        .log(
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts:86]:
```suggestion
        .logDetail(
```

File: packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        );
+    },
+
+    displaySmartSnippetSuggestionsAnswer: (index: number, value: string) => {
+      selector
+        .smartSnippetSuggestionsAnswer(index)
+        .then((elem) => {
+          expect(elem[0].innerHTML).to.eq(value);
+        })
+        .log(
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts:97]:
```suggestion
        .logDetail(
```

File: packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+          `should display the correct  answer of the suggestion at the index ${index}`
+        );
+    },
+
+    displaySmartSnippetSuggestionsSourceUri: (index: number, value: string) => {
+      selector
+        .smartSnippetSuggestionsSourceUri(index)
+        .contains(value)
+        .log(
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts:106]:
```suggestion
        .logDetail(
```

File: packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    },
+
+    displaySmartSnippetSuggestionsSourceTitle: (
+      index: number,
+      value: string
+    ) => {
+      selector
+        .smartSnippetSuggestionsSourceTitle(index)
+        .contains(value)
+        .log(
```
Comment (mmitiche) [inline packages/quantic/cypress/e2e/default-2/smart-snippet-suggestions/smart-snippet-suggestions-expectations.ts:118]:
```suggestion
        .logDetail(
```

**Review by nathanlb — APPROVED** [review]
The system of injecting the actions into the smaller components looks very smooth. Great planning. Fitting it all together almost looks trivial at this point. GG 👍

**Review by SimonMilord — APPROVED** [review]
👍

**Pull request merged**